apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx

```diff
@@ -0,0 +1,262 @@
+      description: 'Success!',
+      disabled: false,
+    },
+    firstTimeSetup: {
+      icon: arrowIcon,
+      title: CHECKLIST_NAMES.adminApi,
+      description: 'Please enable the Admin API to run this check',
+      disabled: false,
+    },
+    invalidServiceAccount: {
+      icon: clockIcon,
+      title: CHECKLIST_NAMES.adminApi,
+      description: 'Awaiting a valid service account install',
+      disabled: true,
+    },
+    error: {
+      icon: errorIcon,
+      title: CHECKLIST_NAMES.adminApi,
+      description: 'Failed to connect to the Admin API - did you enable the api?',
+      disabled: false,
+    },
+  },
+  DataApi: {
+    success: {
+      icon: successIcon,
+      title: CHECKLIST_NAMES.dataApi,
+      description: 'Success!',
+      disabled: false,
+    },
+    firstTimeSetup: {
+      icon: arrowIcon,
+      title: CHECKLIST_NAMES.dataApi,
+      description: 'Please enable the Data API to run this check',
+      disabled: false,
+    },
+    invalidServiceAccount: {
+      icon: clockIcon,
+      title: CHECKLIST_NAMES.dataApi,
+      description: 'Awaiting a valid service account install',
+      disabled: true,
+    },
+    error: {
+      icon: errorIcon,
+      title: CHECKLIST_NAMES.dataApi,
+      description: 'Failed to connect to the Data API - did you enable the api?',
+      disabled: false,
+    },
+  },
+  GA4Properties: {
+    success: {
+      icon: successIcon,
+      title: CHECKLIST_NAMES.ga4Properties,
+      description: 'Success!',
+      disabled: false,
+      checklistUrl: {
+        title: 'Details',
+        url: 'https://analytics.google.com/analytics/web/',
+      },
+    },
+    firstTimeSetup: {
+      icon: clockIcon,
+      title: CHECKLIST_NAMES.ga4Properties,
+      description: `Check will run once the ${CHECKLIST_NAMES.adminApi} is enabled`,
+      disabled: true,
+    },
+    invalidServiceAccount: {
+      icon: clockIcon,
+      title: CHECKLIST_NAMES.ga4Properties,
+      description: 'Awaiting a valid service account install',
+      disabled: true,
+    },
+    adminApiError: {
+      icon: clockIcon,
+      title: CHECKLIST_NAMES.ga4Properties,
+      description: 'Awaiting the admin api to be enabled',
+      disabled: true,
+    },
+    error: {
+      icon: errorIcon,
+      title: CHECKLIST_NAMES.ga4Properties,
+      description: 'There are no properties listed!',
+      disabled: false,
+      checklistUrl: {
+        title: 'Add GA4 properties',
+        url: 'https://analytics.google.com/analytics/web/',
+      },
+    },
+  },
+  Other: {
+    unknown: {
+      icon: errorIcon,
+      title: CHECKLIST_NAMES.unknown,
+      description: 'Unknown error',
+      disabled: true,
+    },
+  },
+};
+
+export const getApiChecklistURLs = (parameters: KeyValueMap, apiError: boolean) => {
+  const projectId = parameters.serviceAccountKey['project_id'];
+  return {
+    adminApi: {
+      title: apiError ? 'Enable Admin API' : 'Details',
```

Comment:
I think "Details" should be lower case. I wonder about "Enable" too? Ditto for down below